plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/application/TemplateViewerFilter.java

```diff
@@ -38,15 +38,19 @@ public boolean select(Viewer viewer, Object parentElement, Object element) {
 		if(!(element instanceof IResource)) {
 			return false;
 		}
+		final String text = filterText.getText();
+		if(StringUtils.isBlank(text)) {
+			return true;
+		}
 		IResource resource = (IResource) element;
-		if(StringUtils.isBlank(filterText.getText())) {
+		if(resource.getName().contains(text)) {
```

Comment:
@adietish just realized that I'm not sure what the correct fix is here.  If the name returns null I would expect errors elsewhere in the code (i.e. label provider) and we we use the name as something displayed to a user.  Also, this is a filter, what should be the return value if the name is null?  Show the template?  We don't have a name to display?  Don't show the template?  It won't appear anyway.  Suggestions?